src/channel/mpmc.rs

```diff
@@ -182,12 +203,55 @@ where
         }
     }
 
+    /// This path should be only used for 0 capacity queues.
+    /// Since the list is not empty, a value is available.
+    /// Extract it from the sender in order to return it
+    fn take_from_sender(&mut self) -> T {
+        debug_assert_eq!(0, self.buffer.capacity());
```

Comment:
Actually the discussion on #12 makes me wonder whether we run into this assert if the producer permits are not used. 

In that case we might also run into this branch even if the channel has non-zero capacity. I have to think futher about that, but for the moment we can leave the assert there. Let me know what your understanding is.

Reply:
Can't say I know. I'll try testing for this behavior.

Reply:
I added some deadlock tests and from what I understand a forgotten send future only eats up one wakeup, but it doesn't seem to do a lot more. Producers and receivers can still continue send & receiving without problem. So I don't think its an issue.